### PR TITLE
resolved APPCLOUD-421 in Jira

### PR DESCRIPTION
**Pull request by madeesha**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 1]
```diff
@@ -29,14 +29,17 @@ jagg.template("index", function(inputs, outputs, jagg) {
...
+
     %>
 
     <script type="text/javascript" xmlns="http://www.w3.org/1999/html" charset="utf-8">
         var applicationList = <%=applications%>;
         var cloudSpecificApplicationRepresentation = "<%=cloudSpecificApplicationRepresentation%>";
         var taggedApplicationsList;
         var searchbox;
+        var searchCriteria = 'all';
+
```
Comment (vinuri) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Remove unwanted lines

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 2]
```diff
@@ -29,14 +29,17 @@ jagg.template("index", function(inputs, outputs, jagg) {
...
-    
+
     %>
 
     <script type="text/javascript" xmlns="http://www.w3.org/1999/html" charset="utf-8">
         var applicationList = <%=applications%>;
         var cloudSpecificApplicationRepresentation = "<%=cloudSpecificApplicationRepresentation%>";
         var taggedApplicationsList;
         var searchbox;
+        var searchCriteria = 'all';
```
Comment (Sithumini) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: @madeesha shall we add this as a constant value?

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 1 above)
Comment (Sithumini) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: @madeesha unnecessary new lines. pls remove

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 3]
```diff
@@ -100,19 +103,20 @@ jagg.template("index", function(inputs, outputs, jagg) {
...
             });
-            if ($("#item_list").children().length < 1) {
+                 if ($("#item_list").children().length < 1 && searchCriteria != 'all') {
                 $("#item_list").append($('<div class="col-centered col-md-4 "><div class="cloud-new-content"> ' +
                  '<h3>Your search did not match any application.</h3></div></div>'));
-            }
-        }
+                 }
+                }
+
```
Comment (Sithumini) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: @madeesha unnecessary newline. pls remove

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 4]
```diff
@@ -100,19 +103,20 @@ jagg.template("index", function(inputs, outputs, jagg) {
                     }
                 }
             });
-            if ($("#item_list").children().length < 1) {
+                 if ($("#item_list").children().length < 1 && searchCriteria != 'all') {
```
Comment (Sithumini) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: @madeesha minor formatting issues

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 2 above)
Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 1 above)
Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 5]
```diff
@@ -100,19 +103,20 @@ jagg.template("index", function(inputs, outputs, jagg) {
                     }
                 }
             });
-            if ($("#item_list").children().length < 1) {
+            if ($("#item_list").children().length < 1 && searchCriteria != 'all') {
```
Comment (amalkasubasinghe) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: you can use the constant here too

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 6]
```diff
@@ -29,14 +29,17 @@ jagg.template("index", function(inputs, outputs, jagg) {
...
+
     %>
 
     <script type="text/javascript" xmlns="http://www.w3.org/1999/html" charset="utf-8">
         var applicationList = <%=applications%>;
         var cloudSpecificApplicationRepresentation = "<%=cloudSpecificApplicationRepresentation%>";
         var taggedApplicationsList;
         var searchbox;
+        var searchCriteria = SEARCH_CRITERIA_ALL;
+
```
Comment (amalkasubasinghe) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: remove new lines

Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 5 above)
Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 3 above)
Comment (madeesha) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Fixed

**Pull request merged**